### DISCUSSION TITLE
Global loading of `R` Packages (for use in all subdocuments)?

### DISCUSSION THREAD

**Discussion by r-cheologist** [quarto-dev/quarto-cli#6877] [Q&A] [answered]
### Description

I'm writing a `book` type document and am wondering what mechanisms (if any) others are using for loading packages such that they are used in all subdocuments and do not to have to explicitly be loaded in each one.

In my current use case that would include

1. The package providing dedicated functions to the document (it itself resides in `inst/document`) and

2. `magrittr`

Any pointers?

**Comment by mcanouil** [comment, marked as answer]
See:

- https://github.com/quarto-dev/quarto-cli/discussions/3250

A bit of a dirty approach as you don't explicitly load in the document is to use `.Rprofile` at the root of your project (alongside `_quarto.yml`).
This has the risk to forget about it and what is done in it, so not really recommended, in addition to the fact that your document is no longer really self-contained in terms of dependency declaration.

**Reply by r-cheologist** [reply]
Indeed. Sorry for the noise. The `.Rprofile` solution may be the least appealing in that thread, but at least 3 options are collected there;

**Reply by mcanouil** [reply]
No need to apologies. ;)

**Reply by cderv** [reply]
> The .Rprofile solution may be the least appealing in that thread, but at least 3 options are collected there;

I would really recommend to use a setup chunk using external R file to load easily in each document. Quarto approach is to consider each .qmd as its own isolated document. it will be executed in a separate session, and so considering that each .qmd should be reproducible on its own seems the best approach to avoid issue. Reusing content through files, packages, or else seems the best approach. 

Hopefully we'll make some improvement in **knitr** to easily reuse a specific chunk from another .Rmd or .qmd. Follow
* https://github.com/yihui/knitr/issues/2041
